### PR TITLE
bazci/githubpost: allow C-Test-Failure to be disabled by CI config

### PR DESCRIPTION
**Pull request by chrisseto**
This commit teaches the GitHub issue logger to respect the new
    `SKIP_LABEL_TEST_FAILURE` environment variable. Setting this envvar to
    any non-zero string value will set `SkipLabelTestFailure` to true, which
    causes the logger to not set the `C-Test-Failure` but continue to log
    failures on aggregate issues.

    Doing so allows us to continue logging the behavioral divergences
    discovered by `TestComposeCompare` without treating them as strict test
    failures.

    The environment variable is intentionally applied to all failures within
    a given CI configuration rather than on a test name basis. This should
    incentivize relegation of high failure tests to isolated CI builds and
    set an appropriate threshold for when a test may be excluded from being
    considered a failure.

Informs: #99181, #89361, #108652, #82867
Release note: None

### CODE REVIEW

**Comment by chrisseto** [comment]
Once this PR is merged, I'll make sure that the ComposeTest CI configurations are updated accordingly 🎉

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109173)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @chrisseto, @DarrylWong, @herkolategan, and @rimadeodhar)

___
*[`pkg/cmd/roachtest/github.go` line 225 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/109173#-NcOidbd73MfWhffJuwp:-NcOidbd73MfWhffJuwq:b-53o4nv) ([raw file](https://github.com/cockroachdb/cockroach/blob/f27bfa4bf88c2deb8d7ddf320e1047e853cdcb52/pkg/cmd/roachtest/github.go#L225)):*
> ```Go
> 		TestName:             issueName,
> 		Message:              issueMessage,
> 		SkipLabelTestFailure: defaultOpts.SkipTestFailureLabel || infraFlake, // infra-flakes or opted-out configurations are not marked as C-test-failure
> ```

i believe this is only used by roachtests

the one used by TestComposeCompare is here: https://github.com/cockroachdb/cockroach/blob/eb78af2db46a49403f98f0a6a8a1be6f2012cd06/pkg/cmd/bazci/githubpost/githubpost.go#L80

(i think)

**Review by chrisseto — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109173)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, @rafiss, and @rimadeodhar)

___
*[`pkg/cmd/roachtest/github.go` line 225 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/109173#-NcOidbd73MfWhffJuwp:-NcOl-o6DRRyZk6siUol:b15zz73) ([raw file](https://github.com/cockroachdb/cockroach/blob/f27bfa4bf88c2deb8d7ddf320e1047e853cdcb52/pkg/cmd/roachtest/github.go#L225)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i believe this is only used by roachtests

the one used by TestComposeCompare is here: https://github.com/cockroachdb/cockroach/blob/eb78af2db46a49403f98f0a6a8a1be6f2012cd06/pkg/cmd/bazci/githubpost/githubpost.go#L80

(i think)
</blockquote></details>

Whoops! Fixed. There's not a great place to slot in a test but it is just a one line change. WDYT?

**Review by rafiss — COMMENTED** [review]
the approach seems good to me! but i think the TestEng team should get final sign-off on the name of the environment variable (e.g., there's a convention to start variables with `COCKROACH_`, or maybe there's a place that variables like this should be documented).

and they also can best say if there's a way to test this. (cc @renatolabs)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109173#-:-NcOnu7_3RRmhwHJLs00:beb4m8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, and @rimadeodhar)

File: pkg/cmd/bazci/githubpost/githubpost.go
Comment (renatolabs) [inline pkg/cmd/bazci/githubpost/githubpost.go:89]: While I don't think this needs a `COCKROACH_` prefix necessarily, I would probably choose an env var name that is closer to the post request option this affects: maybe `SKIP_LABEL_TEST_FAILURE`.

I also think it should be relatively straightforward to add a test case for this in `TestListFailuresFromJSON`.

@cockroachdb/dev-inf would know better if there's a place such env vars are documented; I'm not aware of them being listed anywhere.

**Review by renatolabs — COMMENTED** [review]
Nit: the commit/PR prefix should be `bazci`; roachtest is not being changed in this commit.

File: pkg/cmd/bazci/githubpost/githubpost.go
Comment (chrisseto) [reply pkg/cmd/bazci/githubpost/githubpost.go]: I made some incorrect assumptions about the test case. Unittest added and PR/commit updated to the right package!

**Review by renatolabs — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109173#-:-NcSjfKc8C1DaRdNThew:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @cockroachdb, @DarrylWong, @herkolategan, @rafiss, and @rimadeodhar)

**Comment by rafiss** [comment]
blathers backport 23.1.9-rc

**Comment by rafiss** [comment]
blathers backport 22.2

**Pull request merged**
